### PR TITLE
Split boost dependency out into separate file

### PR DESCRIPTION
**Pull request by sarum9in**
Boost dependency is very complex and it is easier to manage it in a separate file.

### CODE REVIEW

File: mesonbuild/dependencies/boost.py [hunk 1]
```diff
@@ -0,0 +1,687 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This file contains the detection logic for miscellaneous external dependencies.
+
+import glob
+import os
+import re
+import shlex
```
Comment (ignatenkobrain) [inline mesonbuild/dependencies/boost.py]: __[Flake8]__

[F401] 'shlex' imported but unused

File: mesonbuild/dependencies/boost.py [hunk 2]
```diff
@@ -0,0 +1,687 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This file contains the detection logic for miscellaneous external dependencies.
+
+import glob
+import os
+import re
```
Comment (ignatenkobrain) [inline mesonbuild/dependencies/boost.py]: __[Flake8]__

[F401] 're' imported but unused

File: mesonbuild/dependencies/boost.py [hunk 3]
```diff
@@ -0,0 +1,687 @@
...
+import shlex
+
+from .. import mlog
+from .. import mesonlib
+from ..environment import detect_cpu_family
+
+from .base import (
```
Comment (ignatenkobrain) [inline mesonbuild/dependencies/boost.py]: __[Flake8]__

[F401] '.base.PkgConfigDependency' imported but unused

**Comment by sarum9in** [comment]
Removed unused imports.

**Comment by bredelings** [comment]
Splitting this out into a separate file seems like a good idea.

**Pull request merged**